compilers/concrete-compiler/compiler/lib/Common/Keys.cpp

```diff
@@ -28,12 +29,50 @@ Message<ProtoKey> keyToProto(const Key &key) {
   Message<ProtoKey> output;
   auto proto = output.asBuilder();
   proto.setInfo(key.getInfo().asReader());
-  proto.setPayload(vectorToProtoPayload(key.getBuffer()).asReader());
+  proto.setPayload(vectorToProtoPayload(key.getTransportBuffer()).asReader());
   return std::move(output);
 }
 
+void writeSeed(struct Uint128 seed, std::vector<uint64_t> &buffer) {
+  buffer[0] = (uint64_t)seed.little_endian_bytes[0];
+  buffer[0] += (uint64_t)seed.little_endian_bytes[1] << 8;
+  buffer[0] += (uint64_t)seed.little_endian_bytes[2] << 16;
+  buffer[0] += (uint64_t)seed.little_endian_bytes[3] << 24;
+  buffer[0] += (uint64_t)seed.little_endian_bytes[4] << 32;
+  buffer[0] += (uint64_t)seed.little_endian_bytes[5] << 40;
+  buffer[0] += (uint64_t)seed.little_endian_bytes[6] << 48;
+  buffer[0] += (uint64_t)seed.little_endian_bytes[7] << 56;
+  buffer[1] = (uint64_t)seed.little_endian_bytes[8];
+  buffer[1] += (uint64_t)seed.little_endian_bytes[9] << 8;
+  buffer[1] += (uint64_t)seed.little_endian_bytes[10] << 16;
+  buffer[1] += (uint64_t)seed.little_endian_bytes[11] << 24;
+  buffer[1] += (uint64_t)seed.little_endian_bytes[12] << 32;
+  buffer[1] += (uint64_t)seed.little_endian_bytes[13] << 40;
+  buffer[1] += (uint64_t)seed.little_endian_bytes[14] << 48;
+  buffer[1] += (uint64_t)seed.little_endian_bytes[15] << 56;
+}
+
+void readSeed(struct Uint128 &seed, std::vector<uint64_t> &buffer) {
+  seed.little_endian_bytes[0] = buffer[0];
```

Comment:
???
a modulo or and is missing ?

Reply:
c'est une conversion implicite qui le fait ? en explicite c'est mieux

Reply:
yes implicit conversion